### ISSUE TITLE
Nomad Agent Runtime Errors 

### ISSUE THREAD

**Issue by ptarpan** [hashicorp/nomad#2960]
### Nomad version
0.5.5 

### Operating system and Environment details
Windows Server 2012 R2 
### Issue
Nomad agent goes into a panic, throws runtime errors, and takes the node offline.

### Interim Solution
Restarting the agent solves the issue but this is not sustainable.  

### Reproduction steps
- Setup a windows worker 
- Run the parameterized job 
- Dispatch the parameterized job 

### Nomad Server logs (if appropriate)
N/A
### Nomad Client logs (if appropriate)
2017/08/01 18:13:41.795903 [DEBUG] client: state updated to ready
2017/08/01 18:13:55.490793 [DEBUG] client: RPC failed to server 10.131.101.9:4647: rpc error: stream closed
2017/08/01 18:14:24.190181 [DEBUG] client: 9 evaluations triggered by node update
2017/08/01 18:14:24.190181 [DEBUG] client: state updated to ready
2017/08/01 18:15:23.465393 [DEBUG] client: 9 evaluations triggered by node update
2017/08/01 18:15:23.465393 [DEBUG] client: state updated to ready
2017/08/01 18:15:38.171979 [DEBUG] client: 9 evaluations triggered by node update
2017/08/01 18:15:38.171979 [DEBUG] client: state updated to ready
2017/08/01 18:17:47.363210 [DEBUG] client: 9 evaluations triggered by node update
2017/08/01 18:17:47.363210 [DEBUG] client: state updated to ready
2017/08/01 18:18:11.446762 [DEBUG] client: updated allocations at index 210747 (total 8) (pulled 0) (filtered 8)
2017/08/01 18:18:11.688738 [DEBUG] client: allocs: (added 0) (removed 0) (updated 0) (ignore 8)
panic: runtime error: growslice: cap out of range

goroutine 211925 [running]:
fmt.(*fmt).padString(0xc05cd57600, 0xc042499b6f, 0x177ec333c000)
    /opt/go/src/fmt/format.go:110 +0x119
fmt.(*fmt).fmt_s(0xc05cd57600, 0xc042499b6f, 0x177ec333c000)
    /opt/go/src/fmt/format.go:328 +0x68
fmt.(*pp).fmtString(0xc05cd575c0, 0xc042499b6f, 0x177ec333c000, 0x73)
    /opt/go/src/fmt/print.go:433 +0x19f
fmt.(*pp).printArg(0xc05cd575c0, 0x105b900, 0xc0472e99c0, 0x73)
    /opt/go/src/fmt/print.go:664 +0x803
fmt.(*pp).doPrintf(0xc05cd575c0, 0x126df03, 0x6, 0xc050027d80, 0x2, 0x2)
    /opt/go/src/fmt/print.go:998 +0x11ee
fmt.Sprintf(0x126df03, 0x6, 0xc050027d80, 0x2, 0x2, 0xc04d15d9f0, 0x1e0f17c)
    /opt/go/src/fmt/print.go:196 +0x71
github.com/hashicorp/nomad/command/agent.logIndexes(0xc0545159c0, 0x3, 0x3, 0xc042499b63, 0x6, 0xc042499b6f, 0x6, 0x0, 0x1, 0x1, ...)
    /opt/gopath/src/github.com/hashicorp/nomad/command/agent/fs_endpoint.go:965 +0x14f
github.com/hashicorp/nomad/command/agent.blockUntilNextLog.func1(0x1859260, 0xc042947d70, 0xc0478270e0, 0x1a, 0xc04277eff0, 0xc048427c20, 0xc0426dddc0, 0xa, 0xc04
9b63, 0x6, ...)
    /opt/gopath/src/github.com/hashicorp/nomad/command/agent/fs_endpoint.go:924 +0x297
created by github.com/hashicorp/nomad/command/agent.blockUntilNextLog
    /opt/gopath/src/github.com/hashicorp/nomad/command/agent/fs_endpoint.go:942 +0x308
### Job file (if appropriate)
```
job "win-batch" {
    # Specify the region that this job should run in.
    # Regions are defined by the Nomad servers' configuration.
    region      = "us-west-2"
    # Run this job in AWS.
    datacenters = ["us-west-2"]
    # Run this job as a "batch" type. Each job type has different
    # properties.
    type        = "batch"

    constraint {
        attribute   = "${node.class}"
        value       = "win-worker"
    }

    # Define arbitrary k/v pairs.
    meta {
        MODE   = ""
        INPUT  = ""
        ARGS   = ""
    }

    # Make this a parameterized job. A parameterized job is used to
    # encapsulate a set of work that be carried out on various inputs
    # much like a function definition. This job acts as a function to
    # cluster as a whole.
    parameterized {
        payload       = "forbidden"
        meta_required = ["MODE","INPUT"]
        meta_optional = ["ARGS"]
    }

    # Create an individual task (unit of work). 
    task "t1" {
        # Run the job without any resource isolation.
        # This driver must be enabled on the client. In this case,
        # it is.
        driver = "raw_exec"

        # Configuration is specific to each driver.
        config {
            # Call a C# binary.  
            command = "c:\test.exe"
            args    = [
                "${NOMAD_META_MODE}", "${NOMAD_META_INPUT}",
                "${NOMAD_META_ARGS}",
               "-o", "${NOMAD_ALLOC_DIR}/logs",
            ]
        }

        # Specify the resource requirements for this task.
        resources {
            cpu         = 500 # MHz
            memory      = 256 # MB
            network {
                mbits   = 10
            }
        }
    } // END task.t1
}
```

**Comment by dadgar** [comment]
Looks like an out of memory. This is consistent with some other reports around logs.

**Comment by ptarpan** [comment]
Thx for the response @dadgar. Resolved by upgrading the client to v0.6.0.
